near-sdk-sim/README.md

```diff
@@ -26,79 +32,197 @@ near-sdk = { git = "https://github.com/near/near-sdk-rs.git", tag="3.0.0" }
-  pub fn transfer(&self, amount: U128) -> near_sdk::PendingContractTransaction {
-     //...
-  }
+    // if you run `cargo build` without `--release` flag:
+    CONTRACT_WASM_BYTES => "target/wasm32-unknown-unknown/debug/contract.wasm",
 }
 ```
 
-Thus you need to import this new type into your test file like so:
+Note that this means **you must `build` before you `test`!** Since `cargo test` does not re-generate the `wasm` files that your simulation tests rely on, you will need to `cargo build --all --target wasm32-unknown-unknown` before running `cargo test`. If you made contract changes and _you swear it should pass now_, try rebuilding!
+
+Now you can make a function to initialize your simulator:
 
 ```rust
-extern crate token;
-use token::TokenContract;
+use near_sdk_sim::{init_simulator, to_yocto, STORAGE_AMOUNT};
+
+fn init() {
+    // Use `None` for default genesis configuration
+    let root = init_simulator(None);
+
+    let contract = root.create_user(
+        CONTRACT_ID.to_string(), // account id
+        to_yocto("100"),         // initial balance
+    );
+
+    let contract = contract.deploy(
+        &CONTRACT_WASM_BYTES,
+        CONTRACT_ID.to_string(),
+        STORAGE_AMOUNT
+    );
+
+    (root, contract)
+}
 ```
 
-Next you need to import the simulator's init method and some other imports.
+Now you can start writing tests:
 
 ```rust
-use near_sdk_sim::{init_simulator, deploy, call, to_yocto};
+use near_sdk::serde_json::json;
+use near_sdk_sim::DEFAULT_GAS;
+
+#[test]
+fn simulate_some_view_function() {
+    let (root, contract) = init();
+
+    let actual: String = root.view(
+        CONTRACT_ID.to_string(),
+        "view_something".to_string(),
+        json!({
+            "some_param": "some_value".to_string(),
+        }).to_string().into_bytes()
+    ).unwrap_json();
+
+    assert_eq!("expected", actual);
+}
+
+#[test]
+fn simulate_some_change_method() {
+    let (root, contract) = init();
+
+    let result = root.call(
+        CONTRACT_ID.to_string(),
+        "change_something".to_string(),
+        json!({
+            "some_param": "some_value".to_string(),
+        }).to_string().into_bytes()
+        DEFAULT_GAS / 2,
+        0 // attached deposit
+    );
+
+    assert!(result.is_ok());
+}
 ```
 
-This function takes an optional parameter of a genesis configuration. 
-Next you must include the bytes of the contract you want to test.
+
+## Optional macros
+
+The above approach is a good start, and will work even if your Wasm files are compiled from a language other than Rust.
+
+But if your original files are Rust and you want better ergonomics while testing, `near-sdk-sim` provides a nice bonus feature.
+
+`near-sdk-sim` modifies the `near_bindgen` macro from `near-sdk` to create an additional struct+implementation from your contract, with `Contract` added to the end of the name, like `xxxxxContract`. So if you have a contract with `[package].name` set to `token` with this in its `src/lib.rs`:
```

Comment:
```suggestion
`near-sdk-sim` modifies the `near_bindgen` macro from `near-sdk` to create an additional struct and implementation from your contract, with `Contract` added to the end of the name, like `xxxxxContract`. So if you have a contract with `[package].name` set to `token` with this in its `src/lib.rs`:
```

Reply:
You could have a struct for just the data, and the implementation is optional.  In this case the data that the new struct has is the contract's account_id so that when you call methods it knows the receiver.